pos-print/src/test/java/com/clouway/pos/print/persistent/PersistentCashRegisterRepositoryTest.kt

```diff
@@ -24,31 +24,49 @@ class PersistentCashRegisterRepositoryTest {
 
   @Test
   fun happyPath() {
+    val id1 = repository.register(CashRegister("", "any ip 1", "any ip 1", "any description"))
+    val id2 = repository.register(CashRegister("", "any ip 2", "any ip 2", "any description"))
+
     val expected = Lists.newArrayList(
-      CashRegister("any ip 1", "any ip 1", "any description"),
-      CashRegister("any ip 2", "any ip 2", "any description")
+      CashRegister(id1, "any ip 1", "any ip 1", "any description"),
+      CashRegister(id2, "any ip 2", "any ip 2", "any description")
     )
 
-    repository.register(CashRegister("any ip 1", "any ip 1", "any description"))
-    repository.register(CashRegister("any ip 2", "any ip 2", "any description"))
-
     val actual = repository.getAll()
 
     assert(actual.equals(expected))
   }
 
   @Test (expected = DeviceAlreadyExistException::class)
   fun registerAlreadyPresentDevice() {
-    repository.register(CashRegister("any ip", "any ip", "any description"))
-    repository.register(CashRegister("any ip", "any ip", "any description"))
+    repository.register(CashRegister("", "any ip", "any ip", "any description"))
+    repository.register(CashRegister("", "any ip", "any ip", "any description"))
   }
 
   @Test
   fun findBySourceIp() {
-    val expected = CashRegister("sourceIp", "any ip 1", "any description")
-    repository.register(expected)
+    var expected = CashRegister("", "sourceIp", "any ip 1", "any description")
+    val id = repository.register(expected)
+    expected = CashRegister(id, "sourceIp", "any ip 1", "any description")
     val actual = repository.getBySourceIp("sourceIp").get()
 
     assert(actual.equals(expected))
   }
+
+  @Test
+  fun deleteBySourceIp() {
```

Comment:
It's deleted by id now, not by source ip. 

deleteCashRegister
deleteNonExistingCashRegister